test/e2e/scale.go

```diff
@@ -495,3 +596,36 @@ type workResult struct {
 	clusterName string
 	err         any
 }
+
+func modifyMachineDeployments(baseClusterTemplateYAML []byte, count int) []byte {
+	Expect(baseClusterTemplateYAML).NotTo(BeEmpty(), "Invalid argument. baseClusterTemplateYAML cannot be empty when calling modifyMachineDeployments")
+	Expect(count).To(BeNumerically(">=", 0), "Invalid argument. count cannot be less than 0 when calling modifyMachineDeployments")
+
+	objs, err := yaml.ToUnstructured(baseClusterTemplateYAML)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(objs).To(HaveLen(1), "Unexpected number of objects found in baseClusterTemplateYAML")
+
+	scheme := runtime.NewScheme()
+	framework.TryAddDefaultSchemes(scheme)
+	cluster := &clusterv1.Cluster{}
+	Expect(scheme.Convert(&objs[0], cluster, nil)).Should(Succeed())
+	// Verify the Cluster Topology.
+	Expect(cluster.Spec.Topology).NotTo(BeNil(), "Should be a ClusterClass based Cluster")
+	Expect(cluster.Spec.Topology.Workers).NotTo(BeNil(), "ClusterTopology should have exactly one MachineDeployment. Cannot be empty")
+	Expect(cluster.Spec.Topology.Workers.MachineDeployments).To(HaveLen(1), "ClusterTopology should have exactly one MachineDeployment")
+
+	baseMD := cluster.Spec.Topology.Workers.MachineDeployments[0]
+	allMDs := make([]clusterv1.MachineDeploymentTopology, count)
+	for i := 1; i <= count; i++ {
+		md := baseMD.DeepCopy()
+		md.Name = fmt.Sprintf("md-%d", i)
```

Comment:
nit: Let's do the same name trickery for leading zeros that we do for the cluster name